compiler/resolution/wrappers.cpp

```diff
@@ -2398,7 +2403,14 @@ static FnSymbol* promotionWrap(FnSymbol* fn,
 
   PromotionInfo promotion(fn, info, actualIdxToFormal);
 
-  retval = checkCache(promotionsCache, promotion.fn, &promotion.subs);
+  // When inheriting GPU attributes from a variable declaration, bypass the
+  // promotion cache, since the attributes are call-site-specific. For this
+  // reason, later, do not save the generated call to the cache.
```

Comment:
Not entirely following as I consider promotion to be call-site-specific, as well.

OTOH, if we're caching them, maybe that's missing information in my framework. IOW, do we have a single promotion wrapper for:

```
A = A + 1;
B = B + 1;
```

assuming A and B are arrays of same type? Is that what this cache is about?

Reply:
Yes, I believe that's right. We re-use promotion wrappers where possible. We could cache GPU-modified promotion wrappers do, but we'd need to compare their attributes, and that seemed like a challenge best addressed when it becomes a problem.

Reply:
Ok, thanks. I completely agree with that.